x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/MoveToNextStepUpdateTask.java

```diff
@@ -66,9 +69,9 @@ public ClusterState execute(ClusterState currentState) {
         Settings indexSettings = indexMetaData.getSettings();
         LifecycleExecutionState indexILMData = LifecycleExecutionState.fromIndexMetadata(currentState.getMetaData().index(index));
         if (policy.equals(LifecycleSettings.LIFECYCLE_NAME_SETTING.get(indexSettings))
-            && currentStepKey.equals(IndexLifecycleRunner.getCurrentStepKey(indexILMData))) {
+            && currentStepKey.equals(LifecycleExecutionState.getCurrentStepKey(indexILMData))) {
```

Comment:
should we validate the current step key against the policy registry here too? (like we do here https://github.com/elastic/elasticsearch/pull/49891/files#diff-d790e83eaa3919c33fec51b5bf454c8cR104) 
Or are we seeing this cluster update task more of an internal api?

Reply:
I don't think so, I think the validation here (it's not really validation so much as "check-then-set" protection) should be noop-based, meaning that the comparison is not really for validation, it's to avoid things like a previous cluster state operation that changed the index's policy, or a user manually moving the step prior to this task being executed.